### PR TITLE
Moving tro context to avoid prop drilling

### PR DESCRIPTION
**Pull request by Nepomuceno**
### What does this PR do?

Moving to text to acoid prop drilling when changing properties

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you successfully ran tests with your changes locally? Including container validation.
* [x] Have you lint your code locally prior to submission?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### CODE REVIEW

File: client/src/pages/recording-view/hooks/use-spectrogram-context.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+export const SpectrogramContext = createContext<SpectrogramContextProperties>(null);
+
+export function SpectrogramContextProvider({
+  children,
+  type,
+  account,
+  container,
+  filePath,
+  seedValues = {
+    magnitudeMin: -100,
```
Comment (hannahlspencer) [inline client/src/pages/recording-view/hooks/use-spectrogram-context.tsx:40]: Is there any doc anywhere about why these are the seed values?

File: client/src/pages/recording-view/recording-view.tsx [hunk 2]
```diff
@@ -60,6 +48,38 @@ export function RecordingViewPage() {
       setIQData(displayedIQ);
     }
   }, [displayedIQ]);
+  return (
+    <>
+      <Stage width={spectrogramWidth + 110} height={30}>
```
Comment (hannahlspencer) [inline client/src/pages/recording-view/recording-view.tsx:53]: Would these static numbers cause problems with different screen sizes?

File: client/src/pages/recording-view/hooks/use-spectrogram-context.tsx (hunk 1 above)
Comment (Nepomuceno) [reply client/src/pages/recording-view/hooks/use-spectrogram-context.tsx:40]: I am jsut getting the current ones in the current docs.

File: client/src/pages/recording-view/hooks/__tests__/use-spectrogram.test.tsx [hunk 3]
```diff
@@ -83,21 +106,24 @@ describe('test metadata fetch and fft calculation', () => {
...
+        <AllProviders>
+          <SpectrogramContextProvider
+            type={origin.type}
+            account={origin.account}
+            container={origin.container}
+            filePath={origin.file_path}
+            seedValues={seed}
+          >
+            {children}
+          </SpectrogramContextProvider>
```
[the remark this answers is not in the document]
Comment (Nepomuceno) [reply client/src/pages/recording-view/hooks/__tests__/use-spectrogram.test.tsx]: Done :)

File: client/src/pages/recording-view/recording-view.tsx (hunk 2 above)
Comment (Nepomuceno) [reply client/src/pages/recording-view/recording-view.tsx:53]: I know thye have been default in the previous component as logn as we keep the size of the rulers sttic theu are ok but yeah it is something we will need ot review later.

**Pull request merged**